doc/quickstart.rst

```diff
@@ -0,0 +1,488 @@
+- Subfolders for each node (microcontroller)
+- Each node has a ``setup.cpp`` file defining its devices
+
+Let's create a new system called "my-home":
+
+.. code-block:: bash
+
+   mkdir -p ~/iot-systems
+   cd ~/iot-systems
+   create_system_template my-home
+   cd my-home
+
+This creates the system folder with a template configuration.
+
+Configuring System Settings
+---------------------------
+
+Edit the ``system.conf`` file using your favorite editor:
+
+**On Linux:**
+
+.. code-block:: bash
+
+   micro system.conf
+   # or: nano system.conf
+   # or: kate system.conf
+
+**On Windows (WSL):**
+
+.. code-block:: bash
+
+   micro system.conf
+   # or: nano system.conf
+   # or use notepad++: notepad++.exe system.conf
+
+Update the following settings:
+
+.. code-block:: bash
+
+   # Your WiFi network name
+   IOTEMPOWER_AP_NAME="your-wifi-name"
+   
+   # Your WiFi password
+   IOTEMPOWER_AP_PASSWORD="your-wifi-password"
+   
+   # MQTT broker location
+   # If using OpenWRT router with mosquitto:
+   IOTEMPOWER_MQTT_HOST="_gateway"
+   
+   # If running mqtt_starter on your computer, use your computer's IP:
+   # IOTEMPOWER_MQTT_HOST="192.168.1.100"
+   # (replace with your actual IP - shown when mqtt_starter starts)
+
+Save and close the file.
+
+Creating Your First Node
+========================
+
+Create a new node in your system:
+
+.. code-block:: bash
+
+   cd ~/iot-systems/my-home
+   create_node_template living-room-lamp
+   cd living-room-lamp
+
+This creates a new node folder with template files.
+
+.. note::
+   You can also use ``iot menu`` to create nodes interactively. Just run it
+   in your system folder and select **"Create New Node Folder"** from the menu.
+
+Configuring the Node
+--------------------
+
+Check the board type in ``node.conf``:
+
+.. code-block:: bash
+
+   micro node.conf
+
+The default is usually ``wemos_d1_mini``. Change it if you're using a different board:
+
+.. code-block:: bash
+
+   board="wemos_d1_mini"
+   # or: board="nodemcu"
+   # or: board="esp32dev"
+   # or: board="esp32minikit"
+
+Defining Devices in setup.cpp
+-----------------------------
+
+Edit the ``setup.cpp`` file to define your devices:
+
+.. code-block:: bash
+
+   micro setup.cpp
+
+For a simple example with a button and LED, add:
+
+.. code-block:: cpp
+
+   // Button on pin D3
+   input(button1, D3, "released", "pressed").with_debounce(5);
+   
+   // Onboard LED (note: inverted for most ESP boards)
+   output(blue_led, ONBOARDLED).inverted();
+
+Save and close the file.
+
+Flashing Your First Node
+========================
+
+Connect your microcontroller to your computer via USB.
+
+Platform-Specific Serial Flashing
+---------------------------------
+
+**On Linux:**
+
+1. Find your USB device:
+
+   .. code-block:: bash
+
+      ls /dev/ttyUSB*
+      # or: ls /dev/ttyACM*
+
+2. You may need to add yourself to the ``dialout`` group (one-time setup):
+
+   .. code-block:: bash
+
+      sudo usermod -a -G dialout $USER
```

Comment:
this shoudl have happened already and is better done in IoTempower (if forgotten) with iot install --fix-serial - but should happen during installation - under windows, you need to use usbipd (see documentation) and assign the port of the newly connected devices and set the permissions triggered from windows correctly.

I see windows is explained below, so add a forward reference for usbipd here.